lalrpop/src/lr1/lane_table/construct/merge.rs

```diff
@@ -21,45 +20,46 @@ pub struct Merge<'m, 'grammar: 'm> {
     original_indices: Map<StateIndex, StateIndex>,
     clones: Multimap<StateIndex, Vec<StateIndex>>,
     target_states: Vec<StateIndex>,
-    context_sets: ContextSets<'m>,
+    groups: Groups,
+    rows: Map<StateIndex, ContextSet>
 }
 
 impl<'m, 'grammar> Merge<'m, 'grammar> {
     pub fn new(table: &'m LaneTable<'grammar>,
-               unify: &'m mut UnificationTable<StateSet>,
                states: &'m mut Vec<LR1State<'grammar>>,
-               state_sets: &'m mut Map<StateIndex, StateSet>,
-               inconsistent_state: StateIndex)
+               inconsistent_state: StateIndex,
+               rows: Map<StateIndex, ContextSet>)
```

Comment:
Converting it to normal table from `Map` looks like another place for micro optimization